File: backend/models/expense_model.py
from extensions import db

class Expense(db.Model):
    """Expense model for storing user expenses with category and date information."""

    __tablename__ = 'expense'
    id = db.Column(db.Integer, primary_key=True)
    description = db.Column(db.String(200), nullable=False)
    amount = db.Column(db.Float, nullable=False)
    category_id = db.Column(db.Integer, db.ForeignKey('categories.id'), nullable=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'))
    date = db.Column(db.DateTime, server_default=db.func.now())

File: frontend/src/app/models/expense.model.ts
/**
 * Expense Model
 * Represents an expense record in the application
 */
export interface Expense {
  id?: number;
  description: string;
  amount: number;
  category_id?: number;
  user_id?: number;
  date?: Date | string;
  category?: { id: number; name: string };
}

/**
 * Expense Response from API
 */
export interface ExpenseResponse {
  id: number;
  description: string;
  amount: number;
  category_id: number | null;
  user_id: number;
  date: string;
}
